src/main/java/org/jabref/logic/texparser/DefaultTexParser.java

```diff
@@ -0,0 +1,181 @@
+package org.jabref.logic.texparser;
+
+import java.io.IOException;
+import java.io.LineNumberReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.texparser.Citation;
+import org.jabref.model.texparser.TexParser;
+import org.jabref.model.texparser.TexParserResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultTexParser implements TexParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultTexParser.class);
+
+    /**
+     * It is allowed to add new cite commands for pattern matching.
+     */
+    private static final String[] CITE_COMMANDS = new String[] {"cite", "citep", "citet"};
+    private static final String CITE_REGEX = String.format("\\\\(?:%s)\\{([^\\}]*)\\}",
+            String.join("|", CITE_COMMANDS));
+
+    private static final String[] INCLUDE_COMMANDS = new String[] {"include", "includeonly", "input"};
+    private static final String INCLUDE_REGEX = String.format("\\\\(?:%s)\\{([^\\}]*)\\}",
+            String.join("|", INCLUDE_COMMANDS));
+
+    private final BibDatabase masterDatabase;
+
+    public DefaultTexParser(BibDatabase database) {
+        masterDatabase = database;
+    }
+
+    @Override
+    public TexParserResult parse(Path texFile) {
+        return parseTexFiles(new ArrayList<>(Collections.singletonList(texFile)));
+    }
+
+    @Override
+    public TexParserResult parse(List<Path> texFiles) {
+        return parseTexFiles(texFiles);
+    }
+
+    private TexParserResult parseTexFiles(List<Path> texFiles) {
+        TexParserResult result = new TexParserResult(masterDatabase);
+        int fileIndex = 0;
+
+        while (fileIndex < texFiles.size()) {
```

Comment:
Please use a for loop. Instead of updating the `texFiles` with new files, I would prefer a recursive approach: parse the given files and make a list of all referenced latex files, then parse these files (repeating as long as necessary) and finally return the combined results.